simulator/scheduler/plugin/plugins.go

```diff
@@ -271,3 +265,32 @@ func registeredFilterScorePlugins() ([]v1beta2.Plugin, error) {
 
 	return append(registeredscorepls, registeredfilterpls...), nil
 }
+
+// mergeScorePluginWeight merge weight of registered and enabled score plugin.
+func mergeScorePluginWeight(cfg *schedulerConfig.KubeSchedulerConfiguration) (map[string]int32, error) {
+	scorePluginWeight := make(map[string]int32)
+	registeredScorePlugin, err := config.RegisteredScorePlugins()
+	if err != nil {
+		return nil, xerrors.Errorf("get registered score plugins: %w", err)
+	}
+	for _, p := range registeredScorePlugin {
+		scorePluginWeight[p.Name] = 0
+		if p.Weight != nil {
+			scorePluginWeight[p.Name] = *p.Weight
+		}
+	}
+
+	// TODO: support multi-scheduler
+	// overwrite scorePluginWeight by weight of enabled plugin in scheduler-config
+	if len(cfg.Profiles) == 0 {
+		return nil, xerrors.Errorf("get enabled score plugins: %w", err)
+	}
```

Comment:
I believe a given `cfg` should already have the all weight we should use.

Because ↓here merges the default weight and user-specified weight.
https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3c90c5e2aeb7d9e22ed0e8b591b059dea5dced4b/simulator/scheduler/plugin/plugins.go#L174-L184

So, we can eliminate these lines, wdyt?

Reply:
I think you can eliminate other lines(L272 - L281) as well since `cfg` should have all plugins score.
The code I pointed in [the comment](https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/242#discussion_r1003581313) already merges the weights for all plugins (given by users + the default weight in case users don't set any weight for the plugin.)

Reply:
If I eliminate lines(L272 - L281) and user set default weights ↓here for their custom plugins，these weights will not take effect.

> https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/3c90c5e2aeb7d9e22ed0e8b591b059dea5dced4b/simulator/scheduler/config/plugin.go#L56-L60

But I think eliminating L272-L281 is reasonable. Because user should explicitly set weight in scheduler configuraton, not in the code. 

I will remove lines(L272 - L281)